raiden/connection_manager.py

```diff
@@ -163,20 +163,22 @@ def connect(
 
             log_open_channels(self.raiden, self.registry_address, self.token_address, funds)
 
-            qty_network_channels = views.count_token_network_channels(
-                views.state_from_raiden(self.raiden), self.registry_address, self.token_address
-            )
-
-            if qty_network_channels == 0:
+            if not self._have_online_channels_to_connect_to():
```

Comment:
So that would mean that multiple nodes can/might open channels with `BOOTSTRAP_ADDR`

Reply:
@Dominik1999 Is that OK?

Reply:
Why would a node want to be connected to that Address?

Reply:
We cannot find other nodes if they are not part of a channel. That's why the first node used connects to this address.

Now with the filtering this problem might appear at any time, because the nodes get filtered out when offline.

I don't have a better idea a think this is fine for the time being.

Reply:
then any other alternative than `BOOTSTRAP_ADDR` is better. Since the probability of making use of that channel in any time is with `BOOTSTRAP_ADDR` 0 and with the other addresses >0. 

But if you cannot find online nodes to connect with, why don't we tell the user? If there are no online nodes to connect with, there might also not be a lot of fun using Raiden

Reply:
This is how the connection manager is designed to work. I would say that as far as this PR is concerned it should be merged as it solves a real problem.

If you want to change the behaviour of the connection manager make another issue and work on it further.

Reply:
ok, then let's do that. I think we should not use the connection manager if there is no online node to connect to

Reply:
> This is how the connection manager is designed to work. I would say that as far as this PR is concerned it should be merged as it solves a real problem.

@LefterisJP Can we change that so that the second node would not connect to the bootstrap address again, but to any offline node instead?

Reply:
@palango sure that should be easy

Reply:
> @LefterisJP Can we change that so that the second node would not connect to the bootstrap address again, but to any offline node instead?

Done. Check the last 2 commits to this PR.

Unfortunately this logic change made the test a bit more complicated as the test was supposed to test that nobody connects to an offline node. But I think the adjusted test now works. :wink: